examples/app_gradio.py

```diff
@@ -0,0 +1,133 @@
+import PIL.Image
+from vqasynth.datasets import Dataloader
+from vqasynth.embeddings import EmbeddingGenerator
+from PIL import Image
+import os
+from vqasynth.embeddings import TagFilter
+from vqasynth.depth import DepthEstimator
+from vqasynth.localize import Localizer
+from vqasynth.scene_fusion import SpatialSceneConstructor
+from vqasynth.prompts import PromptGenerator
+from datasets import DatasetDict, Dataset
+
+import open3d as o3d
+import os
+import gradio as gr
+import PIL
+
+cache_dir = './vqasynth_output'
+images = 'images'
+
+def resize_image_keep_aspect_ratio(example, images_column, max_width=500):
+        # Get the original image
+        image = example[images_column]
+
+        # Get the original size
+        width, height = image.size
+
+        # Calculate the new height to maintain aspect ratio
+        if width > max_width:
+            new_width = max_width
+            new_height = int((new_width / width) * height)
+            resized_image = image.resize((new_width, new_height), Image.LANCZOS)  # Replaced ANTIALIAS with LANCZOS
+        else:
+            # If width is already less than or equal to max_width, don't resize
+            resized_image = image
+
+        # Replace the image in the example
+        example[images_column] = resized_image
+        return example
+
+def run_vqasynth_pipeline(dataset: DatasetDict):
+
+    # Apply the resizing to your dataset
+    dataset = dataset.map(lambda example: resize_image_keep_aspect_ratio(example, images_column=images))
+
+    embedding_generator = EmbeddingGenerator()
+
+    dataset = dataset.map(lambda example: embedding_generator.apply_transform(example, images))
+
+    tag_filter = TagFilter()
+
+
+    # @title Set tags for filtering
+    include_tags = 'Park,Street,Home,Office,Garden,Playground,Warehouse,Shop,Library,Classroom,Kitchen,Living Room,Bedroom,Bathroom,Garage,Balcony,Alley,Sidewalk,Field,Meadow,Tennis court,Driveway,Lobby,Workshop,Gym' #@param {type:"string"}
+    exclude_tags = 'Aerial Photography,Single Subject,Macro Photography,Microscopic Views,Drawings,Unrealistic CGI,Graphs,Charts,Icons,Logos,Text Overlays,Minimalist,Art,Portraits,Close-up shot,Flat Art,Abstract Paintings,Silhouettes,Monochrome Photography,Stock Photos' #@param {type:"string"}
+
+    include_tags = include_tags.strip().split(",")
+    exclude_tags = exclude_tags.strip().split(",")
+
+    # Extract tags
+    dataset = dataset.map(lambda example: tag_filter.apply_transform(example, include_tags + exclude_tags))
+
+    # Filter by tags
+    dataset_filtered = dataset.filter(
+        lambda example: tag_filter.filter_by_tag(
+            example['tag'], include_tags, exclude_tags
+            )
+        )
+
+
```

Comment:
We could remove these because we don't need to filter the single sample a user uploads